### PR TITLE
Use a #define to avoid the runtime function pointer, while..,

### PR DESCRIPTION
**Pull request by nomis52**
still allowing us to inject a function for testing.

### CODE REVIEW

**Comment by nomis52** [comment]
@joshua-moyerman @peternewman

**Comment by nomis52** [comment]
@nighttrune

**Comment by nomis52** [comment]
@nightrune

File: src/logger.h [hunk 1]
```diff
@@ -34,19 +34,24 @@ extern "C" {
...
- * @param enabled true to enable the logger, false to disable.
+ *   Logging_SendResponse() is called. This can be overridden, see the note
+ *   below.
  * @param max_payload_size The maximum size of the payload to be passed to the
  *   TXFunction. Must be at least 2.
+ *
+ * @note If PIPELINE_TRANSPORT_TX is defined in system_pipeline.h, the macro
+ * will override the tx_cb argument.
+ *
+ * A Logger starts off in the disabled state.
```
Comment (peternewman) [inline src/logger.h]: Does Doxygen work if we pass in two @note's, like two @returns?

**Comment by peternewman** [comment]
LGTM

File: src/logger.h (hunk 1 above)
Comment (nomis52) [inline src/logger.h]: Maybe , but I think we should be using less @note rather than more. 'If everything is important, then nothing is.'

Comment (peternewman) [inline src/logger.h]: "Multiple adjacent \note commands will be joined into a single paragraph. Each note description will start on a new line."

If you don't want it as a note, can you move "A Logger starts off in the disabled state." up to below the @brief with a blank line in between, so it shows in the extended info, or use @post "Starts a paragraph where the postcondition of an entity can be described."

Comment (nomis52) [inline src/logger.h]: I've moved up the "A Logger starts off ... " and removed @note.

**Pull request merged**